SUILesson4/SUILesson4/PlayerViewModel.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  ModelView.swift
+//  SUILesson4
+//
+//  Created by Григоренко Александр Игоревич on 22.01.2023.
+//
+
+import Foundation
+import AVFoundation
+
+/// Модель проигрывателя.
+final class PlayerViewModel: ObservableObject {
+
+    // MARK: - private Constants
+
+    private enum Constants {
+        static let trackFormat = "mp3"
+        static let emptyString = ""
+        static let zeroValue = 0
+        static let floatZeroValue: CGFloat = 0.0
+        static let unoValue = 1
+    }
+
+    // MARK: - public methods
+
+    func play() {
+        guard
+            let songIndex = currentSongIndex
+        else {
+            playSong(name: musicAlbum.first?.name ?? Constants.emptyString)
+            player?.play()
+            currentSongIndex = Constants.zeroValue
+            return }
+
+        playSong(name: musicAlbum[songIndex].name)
+        player?.play()
+    }
+
+    func stop() {
+        player?.stop()
+        currentSongIndex = nil
+    }
+
+    func playNext() {
+        guard
+            let index = currentSongIndex
+        else { return }
+
+        let nextIndex = index + Constants.unoValue
+
+        guard nextIndex != musicAlbum.count else { return }
+
+        currentSongIndex = nextIndex
+
+        player?.stop()
+        playSong(name: musicAlbum[nextIndex].name)
+        player?.play()
+    }
+
+    func playPreviusSong() {
+        guard
+            let index = currentSongIndex
+        else { return }
+
+        let previusIndex = index - Constants.unoValue
+
+        guard previusIndex != Constants.zeroValue else { return }
+
+        currentSongIndex = previusIndex
+
+        player?.stop()
+        playSong(name: musicAlbum[previusIndex].name)
+        player?.play()
+    }
+
+    func setTime(value: Float) {
+        guard let time = TimeInterval(exactly: value) else { return }
+        player?.currentTime = time
+        player?.play()
+    }
+
+    func pause() {
+        player?.pause()
+    }
+
+    func changeVolume(volume: Float) {
+        player?.volume = volume
+    }
+
+    // MARK: - private properties
+
+    @Published private var player: AVAudioPlayer?
+    @Published var currentSongIndex: Int?
+    @Published var musicAlbum = SongAlbum().songs
+    @Published var currentDuratiuon = Double(Constants.zeroValue)
+    @Published private var maxDuration = Constants.floatZeroValue
+    @Published private var volume = Constants.floatZeroValue
+    private var timer: Timer?
```

Comment:
```suggestion

    private var timer: Timer?
```
отделяй враперы от обычных свойств